src/aspire/utils/filters.py

```diff
@@ -29,9 +29,7 @@ def evaluate(self, omega):
         """
         if omega.ndim == 1:
             ensure(self.radial, f'Cannot evaluate a non-radial filter on 1D input array.')
-        elif omega.ndim == 2:
-            ensure(omega.shape[0] == self.dim, f'Omega must be of size {self.dim} x n')
-
+        self.dim = omega.shape[0]
```

Comment:
I don't think you want to do this - the caller passing in an (maybe incorrect) omega shouldn't modify the filter for subsequent calls!

What we were doing originally was to make sure that omega's first dimension will match what this filter expects (as `self.dim`). The only difference now is that if `self.dim` is None, then supposedly the filter will work for any omega, and we should trust that `_evaluate` will do the right thing.

I won't write what I think this line should be, because I think it's constructive to look at the code and try to find this out.

Reply:
If I keep this `elif`,  testIdentityFilter, testScalarFilter, and testZeroFilter will fail because the self.dim is initialized with None, it needs to get the dimension information from somewhere else. Here I get it from the omega.shape[0]. Maybe there is another way? I will take a look again.

Reply:
We can explicitly initialize the dim with the desired value (not None) in a new object for these three classes and keep the elif statement. Which way do you prefer to?

Reply:
@vineetbansal  Added checking consistency of self.dim when it is not none in commit f1e1910.